contracts/Shortfall/Shortfall.sol

```diff
@@ -168,23 +201,30 @@ contract Shortfall is Ownable2StepUpgradeable, AccessControlledV8, ReentrancyGua
         __TokenDebtTracker_init();
         minimumPoolBadDebt = minimumPoolBadDebt_;
         riskFund = riskFund_;
-        waitForFirstBidder = DEFAULT_WAIT_FOR_FIRST_BIDDER;
-        nextBidderBlockLimit = DEFAULT_NEXT_BIDDER_BLOCK_LIMIT;
         incentiveBps = DEFAULT_INCENTIVE_BPS;
         auctionsPaused = false;
+
+        // values are assigned to variables depending on the Mode
+        if (Toggle == Mode.BLOCK_NUMBER) {
```

Comment:
Instead of a `Toggle` variable, we could do this:

* set this contract as `abstract`
* define two new contracts, inheriting this contract: `TimeBasedShortfall` and `BlockNumberBasedShortfall`
* The constants `DEFAULT_WAIT_FOR_FIRST_BIDDER` and `DEFAULT_NEXT_BIDDER_BLOCK_LIMIT` would be defined in `BlockNumberBasedShortfall`
* The constants `DEFAULT_WAIT_FOR_FIRST_BIDDER_IN_SECONDS` and `DEFAULT_NEXT_BIDDER_BLOCK_TIMESTAMP_LIMIT` would be defined in `TimeBasedShortfall`
* in this case, we could rename this `initialize` function to `__Shortfall_init(...)`, and in the `initialize` function of the `BlockNumberBasedShortfall` contract we could have:
```
function initialize(...) {
  __Shortfall_init(...);
  waitForFirstBidder = DEFAULT_WAIT_FOR_FIRST_BIDDER;
  nextBidderBlockLimit = DEFAULT_NEXT_BIDDER_BLOCK_LIMIT;
}
```
* it would be something similar for `TimeBasedShortfall`, but using its own constants